### ISSUE TITLE
Remove old/unused Agents

### ISSUE THREAD

**Issue by BrianJKoopman** [simonsobs/socs#264]
A while ago on a call we identified Agents that have had their functionality replaced and are no longer needed. We should remove them so that we don't need to maintain them (particularly the smurf_recorder, which has tests that depend on so3g.)

I believe these were:
- `agents/smurf_recorder/`
- `agents/pysmurf_archiver/`

I think we could also remove:
- `agents/hwp_sim/`
- `agents/smurf/`

The `smurf_stream_simulator` was used with the smurf-recorder. Is it still useful for other purposes? (My suspicion is yes, but I thought I'd ask.)

**Comment by BrianJKoopman** [comment]
For the record, we discussed this last week and decided the `smurf_stream_simulator` should stick around. And that, while the archiver is still being used some places, it can be removed.
